contracts/upgradeables/games/GUnits.sol

```diff
@@ -372,12 +377,39 @@ contract GUnits is
 
     // @dev Sets the token
     // @param _token The address of the token to set
-    function setToken(address _token) external onlyRole(DEV_CONFIG_ROLE) {
-        if (_token == address(0)) {
+    // @param _previousTokenRecipient The address to send the previous token to
+    function setToken(address _newToken) external whenPaused onlyRole(DEV_CONFIG_ROLE) {
+        if (_newToken == address(0)) {
             revert AddressIsZero();
         }
-        token = _token;
-        emit TokenSet(_token);
+
+        if (totalSupply > 0) {

```

Comment:
just readability if you check the totalSupply && the previousTokenBalance here, you don't need the second if.